src/ir_MilesTag2.cpp

```diff
@@ -0,0 +1,155 @@
+    // Send the supplied data.
+    for (uint64_t mask = 1ULL << (nbits - 1); mask; mask >>= 1)
+      if (data & mask) {  // Send a 1
+        space(kMilesSpace);
+        mark(kMilesOneMark);
+      } else {  // Send a 0
+        space(kMilesSpace);
+        mark(kMilesZeroMark);
+      }
+      space(kMilesRptLength);
+  }
+}
+#endif  // SEND_MILESTAG2
+
+#if DECODE_MILESTAG2
+/// Decode the supplied MilesTag2 message.
+/// Status: NEEDS TESTING
+/// @param[in,out] results Ptr to the data to decode & where to store the result
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return True if it can decode it, false if it can't.
+/// @see https://github.com/crankyoldgit/IRremoteESP8266/issues/706
+bool IRrecv::decodeMiles(decode_results *results, uint16_t offset,
+                        const uint16_t nbits, const bool strict) {
+  /*
+  uint16_t gap_pos = 0;
+  // we got alot more data than we thought, let's find last GAP and work from it
+  if (results->rawlen >= (2 * nbits + 1)) 
+  {
+      for (uint16_t ind = 0 ; ind < results->rawlen; ind++)
+      {
+          if (matchAtLeast(*(results->rawbuf+ind),34000)) gap_pos = ind;
+      }
+  }
+
+  if (results->rawlen>gap_pos) offset = gap_pos+1;
+  */  
+  // Compliance
+  if (strict) {
+    switch (nbits) {  // Check we've been called with a correct bit size.
+      case 14:
+      case 24:
+        break;
+      default:
+        DPRINT("incorrect nbits:");
+        DPRINTLN(nbits);
+        return false;  // The request doesn't strictly match the protocol defn.
+    }
+  }
+  uint64_t data = 0;
+  // Header
+  if (!matchMark(*(results->rawbuf + offset++),
+                  kMilesHdrMark, kUseDefTol, kMarkExcess)) return 0;
+  // Data
+  uint16_t shift = 0;
+  for (shift = 0; shift < nbits * 2; shift += 2) {
+    // Is the bit a '1'?
+    if (matchMark(*(results->rawbuf + 1 + offset + shift),
+                 kMilesOneMark, kUseDefTol, kMarkExcess) &&
+        matchSpace(*(results->rawbuf + offset + shift),
+                  kMilesSpace, kUseDefTol, kMarkExcess)) {
+      data = (data << 1) | 1;
+    } else if (matchMark(*(results->rawbuf + 1 + offset + shift),
+                        kMilesZeroMark, kUseDefTol, kMarkExcess) &&
+               matchSpace(*(results->rawbuf + offset + shift),
+                         kMilesSpace, kUseDefTol, kMarkExcess)) {
+      data <<= 1;  // The bit is a '0'.
+    } else {
+      return false;  // It's neither, so fail.
+    }
+  }
+  // Success
+  results->bits = nbits;
+  results->value = data;
+  switch (nbits) {
+    case 14:
+      results->decode_type = decode_type_t::MILESTAG2SHOT;
+      break;
+    case 24:
+      results->decode_type = decode_type_t::MILESTAG2MSG;
+      break;
+    default:
+      return false;
+  }
```

Comment:
I think this is a bad idea. The two different protocols are really just the same, but with different bit lengths.
e.g. Similar to `SONY`, which has three different bit lengths for the same Protocol (name).

Let's just make it a single protocol number/type of `decode_type_t::MILESTAG2`.